go/pkg/cosmos/history.go

```diff
@@ -49,6 +49,27 @@ func NewDefaultSources(client *HTTPClient, pubkey string, formatTx func(*coretyp
 	}
 }
 
+func ValidatorHistorySources(client *HTTPClient, pubkey string, formatTx func(*coretypes.ResultTx) (*Tx, error)) map[string]*TxState {
+	request := func(query string, page int, pageSize int) ([]HistoryTx, error) {
```

Comment:
dupe request function logic if you want to DRY up at all

Reply:
I give up on this one. I've tried rewriting `TxHistorySources` in a few different ways but they all lead to nowhere but it's super hard to get the big picture when everything is called a `request` and defined in an anonymous nested lambda. I'm strongly against using anonymous fn's for anything but the simplest operations. I wanted to unwrap this function into separate bits and couldn't really get there

Reply:
Imo this really needs a more comprehensive rewrite, since the logic is split all over the place

Reply:
Yeah, one of the most complicated parts of the codebase thus far. I am down to brainstorm some ideas with you or will do a follow up to try and address the situation a bit better.

Reply:
sounds great, let's give it a shot